dbt_sugar/core/connectors/base.py

```diff
@@ -17,34 +18,84 @@ class BaseConnector(ABC):
     Parent class of all the connectors.
     """
 
-    @abstractmethod
-    def generate_connection(self) -> sqlalchemy.engine:
+    def __init__(
+        self,
+        connection_params: Dict[str, str],
+    ) -> None:
         """
-        Method that creates the connection.
+        Init method to instanciate the credentials.
 
-        Returns:
-            sqlalchemy.engine: Engine to connect to the database.
+        Args:
+            connection_params (Dict[str, str]): Dict with connection parameters.
         """
-        logging.info("Genetating connection")
+        self.connection_url = sqlalchemy.engine.url.URL(
+            drivername="postgresql+psycopg2",
```

Comment:
I don't understand why be base is using a postrgres connector? To me the base shouldn't implement any of the connection URL stuff that's a responsibility of the concrete classes